pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/design/NcssTypeCountRule.java

```diff
@@ -10,7 +10,9 @@
  * Non-commented source statement counter for type declarations.
  *
  * @author ported from Java original of Jason Bennett
+ * @deprecated Since 7.19.0. Use the rule {@link NcssCountRule} instead.
  */
+@Deprecated
```

Comment:
Also deprecate the corresponding test classes (NcssTypeCountTest, NcssMethodCountTest, NcssConstructorCountTest, ExcessiveClassLengthTest)